imod/mf6/npf.py

```diff
@@ -461,3 +469,74 @@ def _validate(self, schemata, **kwargs):
 
     def get_regrid_methods(self) -> Optional[dict[str, Tuple[RegridderType, str]]]:
         return self._regrid_method
+
+    @classmethod
+    def from_imod5_data(
+        cls,
+        imod5_data: dict[str, dict[str, GridDataArray]],
+        target_grid: GridDataArray,
+        regridder_types: Optional[dict[str, tuple[RegridderType, str]]] = None,
+    ) -> "NodePropertyFlow":
+        """
+        Construct an npf-package from iMOD5 data, loaded with the
+        :func:`imod.formats.prj.open_projectfile_data` function.
+
+        .. note::
+
+            The method expects the iMOD5 model to be fully 3D, not quasi-3D.
+
+        Parameters
+        ----------
+        imod5_data: dict
+            Dictionary with iMOD5 data. This can be constructed from the
+            :func:`imod.formats.prj.open_projectfile_data` method.
+        target_grid: GridDataArray
+            The grid that should be used for the new package. Does not
+            need to be identical to one of the input grids.
+        regridder_types: dict, optional
+            Optional dictionary with regridder types for a specific variable.
+            Use this to override default regridding methods.
+
+        Returns
+        -------
+        Modflow 6 npf package.
+
+        """
+
+        data = {
+            "k": imod5_data["khv"]["kh"],
+        }
+        has_vertical_anisotropy = (
+            "kva" in imod5_data.keys()
+            and "vertical_anisotropy" in imod5_data["kva"].keys()
+        )
+        has_horizontal_anisotropy = "ani" in imod5_data.keys()
+
+        if has_vertical_anisotropy:
+            data["k33"] = data["k"] * imod5_data["kva"]["vertical_anisotropy"]
+        if has_horizontal_anisotropy:
+            if not np.all(np.isnan(imod5_data["ani"]["factor"].values)):
+                factor = imod5_data["ani"]["factor"]
+                factor = fill_missing_layers(factor, target_grid, 1)
+                data["k22"] = data["k"] * factor
+            if not np.all(np.isnan(imod5_data["ani"]["angle"].values)):
+                angle1 = imod5_data["ani"]["angle"]
+                angle1 = 90.0 - angle1
```

Comment:
Just a note, maybe you're aware already: iMOD5 chooses the north (or 90 degrees with x-axis) as the principal direction of the horizontal conductivity. MODFLOW 6 chooses the x-axis (0 degrees) as the principal direction (as you'd expect from the unit circle). If I recall correctly, in the iMOD5 conversion, this results in there always being an 90 degree angle defined even if there's no anisotropy. However, this makes the formulation step of MODFLOW 6 more expensive, because it requires trigonometry evaluations. It's worth considering, I think, to consistently transform the iMOD5 conductivity to an x-axis aligned one for k11.

Reply:
@luitjansl could you post an issue for this? Probably worth further investigating if MODFLOW 6 computations become unbearable.